File: tests/fixtures/semgrep-offline/positive/vulnerable.js
/* Every JavaScript rule in the offline profile must flag something here. */
const child_process = require('child_process');
const crypto = require('crypto');

function commandInjection(cmd) { child_process.exec(cmd); }
function codeInjection(input) { return eval(input); }
function dynamicCode(src) { return new Function(src); }
function weakHash(data) { return crypto.createHash('md5').update(data); }
function predictable() { return Math.random(); }
// The env-var form is njsscan's (D12); ours is the per-request half.
function disableTls(url) { return https.request(url, {rejectUnauthorized: false}); }
function htmlSink(el, value) { el.innerHTML = value; }

File: tests/fixtures/semgrep-offline/positive/vulnerable.go
package main

import (
	"crypto/des"
	"crypto/md5"
	"crypto/rc4"
	"crypto/tls"
	"math/rand"
	"os/exec"
)

func commandInjection(userInput string) { exec.Command("sh", "-c", userInput) }
func weakHash(data []byte) []byte       { h := md5.New(); return h.Sum(data) }
func weakCipher(key []byte)             { des.NewCipher(key); rc4.NewCipher(key) }
func tlsDisabled() *tls.Config          { return &tls.Config{InsecureSkipVerify: true} }
func predictable() int                  { return rand.Intn(100) }
